website/blog/2024-04-26-deep-learning-part-2/index.md

```diff
@@ -0,0 +1,287 @@
+slug: deep-learning-part-2
+authors: [puneet-jain, avinash-sooriyarachchi, abe-omorogbe, ben]
+tags: [Deep Learning]
+thumbnail: img/blog/dl-blog-2.png
+---
+
+In the realm of deep learning, finetuning of pre-trained Large Language Models (LLMs) on private datasets is an excellent customization
+option to increase a model’s relevancy for a specific task. This practice is not only common, but also essential for developing specialized
+models, particularly for tasks like text classification and summarization.
+
+In such scenarios, tools like MLflow are invaluable. Tracking tools like MLflow help to ensure that every aspect of the training
+process - metrics, parameters, and artifacts - are reproducibly tracked and logged, allowing for the analysis, comparison, and sharing of tuning iterations.
+
+<!-- truncate -->
+
+In this blog post, we are going to be using [MLflow 2.12](https://mlflow.org/releases/2.12.1) and the
+[recently introduced MLflow Deep Learning features](https://mlflow.org/blog/deep-learning-part-1) to track all the important aspects of fine
+tuning a large language model for text classification, including the use of automated logging of training checkpoints in order to simplify
+the process of resumption of training.
+
+## Use Case: Fine Tuning a Transformer Model for Text Classification
+
+The example scenario that we're using within this blog utilizes the [unfair-TOS](https://huggingface.co/datasets/coastalcph/lex_glue/viewer/unfair_tos) dataset.
+
+In today’s world, it’s hard to find a service, platform, or even a consumer good that doesn’t have a legally-binding terms of service connected
+with it. These encyclopedic size agreements, filled with dense legal jargon and sometimes baffling levels of specificity, are so large that
+most people simply accept them without reading them. However, reports have indicated over time that occasionally, some suspiciously unfair
+terms are embedded within them.
+
+Addressing unfair clauses in Terms of Service (TOS) agreements through machine learning (ML) is particularly relevant due to the pressing
+need for transparency and fairness in legal agreements that affect consumers. Consider the following clause from an example TOS
+agreement: **"We may revise these Terms from time to time. The changes will not be retroactive, and the most current version of the Terms, which will always..."**
+This clause stipulates that the service provider may suspend or terminate the service at any time for any reason,
+with or without notice. Most people would consider this to be quite unfair.
+
+While this sentence is buried quite deeply within a fairly dense document, an ML algorithm is not burdened by the exhaustion that a human
+would have for combing through the text and identifying clauses that might seem a bit unfair. By automatically identifying potentially
+unfair clauses, transformers can help protect consumers from exploitative practices, ensure greater compliance with legal standards,
```

Comment:
nit: `transformers can help protect customers...` sounds a bit weird protect consumers: sounds a bit weird as transformers us a framework not a model, like we don't say `python can help....`. Can we rephrase it to "the ML model can help ..."?